src/pages/TermsOfServiceAcceptance.js

```diff
@@ -23,6 +23,12 @@ export default function TermsOfServiceAcceptance(props) {
     const user = await Storage.getCurrentUser();
     await TosService.acceptTos();
     await Storage.setUserIsLogged(true);
+
+    if (user.isResearcher && !researcherProfileComplete(user)) {
+      history.push('/profile');
+      return;
+    }
```

Comment:
I don't think we need to check for `user.isResearcher` here - all users are always researchers (that's our default user role).

Reply:
I'm on the fence about the `researcherProfileComplete` check. I'm leaning to sending the user to the profile page whenever they accept ToS, regardless of completeness.

1. If a user has never selected an institution, both approaches work
2. If a user has previously filled in an institution, then rejected ToS, I think we want to send them to the profile page so they can see the on-boarding form again if they re-accept ToS. In this case, the current approach doesn't work.

Reply:
I don't have a strong opinion here, but I think when I use other websites, I would expect to be redirected to whatever my main home view would be when accepting the ToS. Like, when I've already had to do an extra unexpected click, I just want to see the content, and this extra profile step would be only mildly annoying (and hopefully only seen once :P)